packages/core/core-flows/src/order/workflows/get-order-detail.ts

```diff
@@ -0,0 +1,159 @@
+import { OrderDTO, OrderDetailDTO } from "@medusajs/types"
+import { deduplicate } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+
+enum PaymentStatus {
+  NOT_PAID = "not_paid",
+  AWAITING = "awaiting",
+  CAPTURED = "captured",
+  PARTIALLY_CAPTURED = "partially_captured",
+  PARTIALLY_REFUNDED = "partially_refunded",
+  REFUNDED = "refunded",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+enum FulfillmentStatus {
+  NOT_FULFILLED = "not_fulfilled",
+  PARTIALLY_FULFILLED = "partially_fulfilled",
+  FULFILLED = "fulfilled",
+  PARTIALLY_SHIPPED = "partially_shipped",
+  SHIPPED = "shipped",
+  PARTIALLY_RETURNED = "partially_returned",
+  RETURNED = "returned",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+export const getOrderDetailWorkflowId = "get-order-detail"
+export const getOrderDetailWorkflow = createWorkflow(
+  getOrderDetailWorkflowId,
+  (
+    input: WorkflowData<{ fields: string[]; order_id: string }>
+  ): WorkflowData<OrderDetailDTO> => {
+    const fields = transform(input, ({ fields }) => {
```

Comment:
Should this happen in the HTTP layer instead? Prob. in the middleware

Reply:
This is a read only workflow where we have to make sure a couple of relations are always present. overall I don't like this flow, nor the aggregations it does, I am just trying to keep the fields we have in v1 for now.